src/types.js

```diff
@@ -49,8 +68,13 @@ const types = {
         }
       }
     }
-    return Promise.resolve(types.strategies[context.def.type](context))
-      .then(res => res === undefined ? context.value : res)
+    // Ensure subtype
+    if (!types.strategies[context.def.type][context.def.sub]) {
+      throw new Error(`Type '${context.def.type}:${context.def.sub}' does not exist`)
+    } else {
+      return Promise.resolve(types.strategies[context.def.type][context.def.sub](context))
```

Comment:
I'm kind of amazed eslint didn't choke on this! You don't need the `else` here-- the return can happen after the `if` unconditionally since you're throwing.